src/WebJobs.Script.WebHost/FileMonitoringService.cs

```diff
@@ -118,6 +109,29 @@ private void InitializeFileWatchers()
                     .Subscribe((msg) => ScheduleRestartAsync(false)
                     .ContinueWith(t => _logger.LogCritical(t.Exception.Message),
                         TaskContinuationOptions.OnlyOnFaulted | TaskContinuationOptions.ExecuteSynchronously)));
+
+            // Delay starting up for logging and debug file watchers to avoid long start up times
+            Utility.ExecuteAfterDelay(InitializeNonBlockingFileWatchers, TimeSpan.FromSeconds(_delayedWatchersDelaySeconds));
+        }
+
+        /// <summary>
+        /// Initializes the file and directory monitoring that does not need to happen as part of a Host startup
+        /// These watchers can be started after a delay to avoid startup performance issue
+        /// </summary>
+        private void InitializeNonBlockingFileWatchers()
+        {
+            if (!_disposed)
```

Comment:
Doesn't this open us up to race conditions now - if the host is disposing but _disposed not yet set?

Reply:
Yes, you are right about this. And I was thinking of having a disposal lock similar to how some other classes have configured. I will add that